thesis-gui/src/components/quiz/Quiz.tsx

```diff
@@ -71,18 +71,21 @@ export const Quiz: React.FC = () => {
 
     const getNewQuestion = () => {
         const randomCategory = getRandom()!;
-
         return questionService.get(randomCategory.id)
             .then(res => {
                 console.log(`Correct answer: ${res.correct.name}`)
-
                 setCategory(randomCategory);
                 setQuestion(res);
                 setSelected(undefined);
                 setQuestionNumber(prev => prev + 1);
-            })
-            .catch(err => console.log(err))
-            .finally(() => setShowResult(false));
+                setShowResult(false)
+            }
+            )
+            .catch(err =>
+            {
+            console.log(err);
+            setTimeout(getNewQuestion, 100);
```

Comment:
Fronted is not my specialization but I want to ask, what is the purpose of waiting 100 milliseconds?

Reply:
Didn't check enough, waiting is not needed in this case